### PR TITLE
CTCTRADER-144 - Adding in multiple incidents

### PR DESCRIPTION
**Pull request by garethgooch**
Draft PR including first round of event answers

### CODE REVIEW

File: test/queries/EventsQuerySpec.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.scalacheck.Arbitrary.arbitrary
+import pages.behaviours.PageBehaviours
+import play.api.libs.json.{JsObject, JsString, Json}
+
+class EventsQuerySpec extends PageBehaviours {
+
+  "AddEventPage" - {
+
+    implicit val asdf: Arbitrary[JsObject] = Arbitrary {
+      for {
```
Comment (kenneth-rayner) [inline test/queries/EventsQuerySpec.scala]: val name??

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -77,8 +79,15 @@ class CheckYourAnswersController @Inject()(override val messagesApi: MessagesApi
...
 
     Seq(mrn, goodsLocation, traderDetails, notificationDetails, events)
   }
+
+  private def eventList(userAnswers: UserAnswers): Seq[SummaryList.Row] = {
+    val numberOfEvents = userAnswers.get(DeriveNumberOfEvents).getOrElse(0)
+    val cyaHelper      = new AddEventsHelper(userAnswers)
+
+    (0 to numberOfEvents).flatMap(cyaHelper.cyaListOfEvent)
+  }
```
[the remark this answers is not in the document]
Comment (ashwinirudrappa) [reply app/controllers/CheckYourAnswersController.scala]: We will look into it later as a separate task

File: test/controllers/events/AddEventControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        "radios" -> Radios.yesNo(form("value"))
+      )
+
+      templateCaptor.getValue mustEqual "events/addEvent.njk"
+      jsonCaptor.getValue must containJson(expectedJson)
+
+      application.stop()
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
```
[the remark this answers is not in the document]
Comment (ashwinirudrappa) [reply test/controllers/events/AddEventControllerSpec.scala]: will discuss with team

File: test/queries/EventsQuerySpec.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import org.scalacheck.Arbitrary
+import org.scalacheck.Arbitrary.arbitrary
+import pages.behaviours.PageBehaviours
+import play.api.libs.json.{JsObject, JsString, Json}
+
+class EventsQuerySpec extends PageBehaviours {
+
+  "AddEventPage" - {
+
+    implicit val event: Arbitrary[JsObject] = Arbitrary {
```
Comment (amaalali) [inline test/queries/EventsQuerySpec.scala]:
```suggestion
    implicit val arbitraryJsObject: Arbitrary[JsObject] = Arbitrary {
```

**Pull request merged**